### ISSUE TITLE
Vector Line Segment Eraser Crash 

### ISSUE THREAD

**Issue by melieconiek** [tahoma2d/tahoma2d#1365]
### Bug Description

Moving to another drawing while the vector eraser in `segment` mode is held down will crash Tahoma.

### Steps to reproduce

1. Create a level with two drawings on the xsheet.
2. Mouse down with the eraser tool in segment mode. (Hold it down.)
3. Use a keybind to move to the next drawing.
**Tahoma will crash. It shouldn't do this.**

### Software Version

1.4

### Operating System

Windows 10

**Comment by artisteacher** [comment]
I might be doing this wrong, but I'm not getting a crash on macOS so far. I am using a recent nightly build, in case it makes a difference.

**Comment by beeheemooth** [comment]
@artisteacher Segment Eraser also works for me (win 8.1), but the crash occurs with Freehand Eraser following the steps described. Can you confirm this on Mac?

**Comment by manongjohn** [comment]
I can't seem to duplicate this.  Can you provide the crash log so I can see where it is crashing?

**Comment by artisteacher** [comment]
Not seeing a crash with freehand erase either. Does the level type matter?

**Comment by manongjohn** [comment]
Ok..with the crash log and dump (which really helped!) that @melieconiek provided in Discord, I discovered why this happens and why I couldn't reproduce it at first.

To reproduce the crash I did the following:
1. Use Normal eraser to erase a portion of any line
2. Switch to Segment eraser
3. Start Segment erasing and without lifting pen/releasing mouse button, use Next Drawing.  Will lead to crash.

The Normal eraser internally keeps track of vectors for Undo purposes, but Segment doesn't. The undo wasn't cleared when normal erasing was complete so when Segment was in progress and switched to a new image, it tried to handle the tracking information again and crashed when it tried to record an Undo operation that wasn't properly initialized.
